src/main/java/org/example/springbootintro/service/book/BookServiceImpl.java

```diff
@@ -35,4 +35,24 @@ public BookDto findById(Long id) {
                 .map(bookMapper::toDto)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by id: " + id));
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookExist(id);
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        bookExist(id);
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    private void bookExist(Long id) {
+        if (bookRepository.findById(id).isEmpty()) {
```

Comment:
we can make our db request lighter since we do not use the book data that we fetch - we can just use `repo.existsById()` instead of `find` which will just count the resulting rows making a request a bit faster
```suggestion
        if (bookRepository.existsById()) {
```